streams/shared/src/main/scala/zio/stream/experimental/ZSink.scala

```diff
@@ -1197,6 +1203,22 @@ object ZSink {
       new ZSink(ZChannel.unwrap(ZIO.access[R](f(_).channel)))
   }
 
+  def utfDecode[Err]: ZSink[Any, Err, Byte, Err, Byte, Option[String]] = {
+    def prepend(bytes: Chunk[Byte]): ZChannel[Any, Err, Chunk[Byte], Any, Err, Chunk[Byte], Any] =
+      ZChannel.write(bytes) *> ZChannel.identity[Err, Chunk[Byte], Any]
+
+    ZSink.take[Err, Byte](4).flatMap { bytes =>
+      bytes.toList match {
```

Comment:
We could skip the conversion to list and back to chunk and work directly on chunks (I know the 1.0 implementation  is like this too, so consider it as an optional optimization - we can also do it later)